stdlib/public/core/UnsafeBufferPointer.swift.gyb

```diff
@@ -366,6 +389,141 @@ public struct Unsafe${Mutable}BufferPointer<Element>
+  ///
+  /// The buffer’s memory must be initialized or the buffer's `Element`
+  /// must be a trivial type. 
+  ///
+  /// - Parameters:
+  ///   - repeatedValue: The instance to assign this buffer's memory to.
+  ///
+  /// Warning: All buffer elements must be initialized before calling this. 
+  /// Assigning to part of the buffer must be done using the `assign(repeating:count:)`` 
+  /// method on the buffer’s `baseAddress`. 
+  @_inlineable
+  public func assign(repeating repeatedValue: Element) {
+    guard let dstBase = _position else {
+      return
+    }
+
+    dstBase.assign(repeating: repeatedValue, count: _end! - dstBase)
+  }
+  
+% end
+
+  /// Executes the given closure while temporarily binding the memory referenced 
+  /// by this buffer to the given type.
+  ///
+  /// Use this method when you have a buffer of memory bound to one type and
+  /// you need to access that memory as a buffer of another type. Accessing
+  /// memory as type `T` requires that the memory be bound to that type. A
+  /// memory location may only be bound to one type at a time, so accessing
+  /// the same memory as an unrelated type without first rebinding the memory
+  /// is undefined.
+  ///
+  /// The entire region of memory referenced by this buffer must be initialized.
+  /// 
+  /// Because this buffer's memory is no longer bound to its `Element` type
+  /// while the `body` closure executes, do not access memory using the
+  /// original buffer from within `body`. Instead, use the `body` closure's
+  /// buffer argument to access the values in memory as instances of type
+  /// `T`.
+  ///
+  /// After executing `body`, this method rebinds memory back to the original
+  /// `Element` type.
+  ///
+  /// - Parameters:
+  ///   - type: The type to temporarily bind the memory referenced by this
+  ///     buffer. The type `T` must be layout compatible
+  ///     with the pointer's `Element` type. The type `T` does not have to be 
+  ///     the same size as the buffer’s `Element` type.
+  ///   - body: A closure that takes a ${Mutable.lower()} typed buffer to the
+  ///     same memory as this buffer, only bound to type `T`. The buffer argument 
+  ///     contains as many complete instances of `T` as can fit within the 
+  ///     region of memory referenced by the original buffer. The closure's
+  ///     pointer argument is valid only for the duration of the closure's
+  ///     execution. If `body` has a return value, that value is also used as
+  ///     the return value for the `withMemoryRebound(to:_:)` method.
+  /// - Returns: The return value, if any, of the `body` closure parameter.
+  ///
+  /// Warning: Unlike this method, the destination type `T` of 
+  /// `Unsafe${Mutable}Pointer.withMemoryRebound(to:capacity:_:)` requires `T` 
+  /// to be the same size as its `Pointee` type.
+  @_inlineable
+  public func withMemoryRebound<T, Result>(
+    to type: T.Type, _ body: (${Self}<T>) throws -> Result) rethrows -> Result {
```

Comment:
Formatting nit: typically, in stdlib style, when the parameters start on another line, the closing parenthesis is also on a separate line like so:

```swift
@_inlineable
public func withMemoryRebound<T, Result>(
  to type: T.Type, _ body: (${Self}<T>) throws -> Result
) rethrows -> Result {
  // ...
}
```